internal/manifests/manifests.go

```diff
@@ -260,28 +266,18 @@ func (m *Manifests) UpdateClusterManifestInternal(ctx context.Context, params op
 		}
 	}
 
-	err = m.uploadManifest(ctx, content, params.ClusterID, destPath)
+	err = m.uploadManifest(ctx, content, params.ClusterID, destPath, constants.ManifestSourceUserSupplied)
 	if err != nil {
 		return nil, err
 	}
 
-	isCustomManifest, err := m.IsUserManifest(ctx, params.ClusterID, srcFolder, srcFileName)
```

Comment:
This seems to suggest that this function could have been used for either user or system manifests. Are we sure that that's not the case now?

Reply:
This is within a function named `UpdateClusterManifestInternal` which, at the time of writing is only called from the user facing API, there are no other calls.

There is no use case for the system to re-upload a manifest for which it already knows the content. 

Therefore we can conclude that any attempt to customize a manifest should render it "user generated"

There is perhaps a thin argument to suggest that we should check the origin of the original manifest before updating it in order to refuse to overwrite a manifest that is "system generated", if the user wishes to override system generated content, they should use the patch manifest system.

Though I did not go to this length as I felt it was overkill.

TLDR: I don't see a problem with the above.